### PR TITLE
Restore encapsulation of distributions layout, broken by gas optimisa…

### PR DESCRIPTION
**Pull request by hoytech**
…tions

### CODE REVIEW

File: src/BaseRewardStreams.sol
Comment (hoytech) [reply src/BaseRewardStreams.sol:371]: OK, but I'll wait until the readability PR lands, because I foresee merge conflicts :)

Comment (kasperpawlowski) [reply src/BaseRewardStreams.sol:371]: the readability PR is merged

Comment (hoytech) [reply src/BaseRewardStreams.sol]: I rebased, and moved the function

**Review by kasperpawlowski — APPROVED** [review]
lgtm

**Pull request merged**
